src/status_im/ui/screens/chat/components/input.cljs

```diff
@@ -73,27 +77,102 @@
-   [rn/text-input {:style                  (styles/text-input)
-                   :ref                    text-input-ref
-                   :maxFontSizeMultiplier  1
-                   :accessibility-label    :chat-message-input
-                   :text-align-vertical    :center
-                   :multiline              true
-                   :default-value          text-value
-                   :editable               (not cooldown-enabled?)
-                   :blur-on-submit         false
-                   :auto-focus             false
-                   :on-focus               #(set-active-panel nil)
-                   :on-change              #(on-text-change (.-text ^js (.-nativeEvent ^js %)))
-                   :max-length             chat.constants/max-text-size
-                   :placeholder-text-color (:text-02 @colors/theme)
-                   :placeholder            (if cooldown-enabled?
-                                             (i18n/label :cooldown/text-input-disabled)
-                                             (i18n/label :t/type-a-message))
-                   :underlineColorAndroid  :transparent
-                   :auto-capitalize        :sentences}]])
+(defn text-input
+  [{:keys [cooldown-enabled? text-value on-text-change set-active-panel text-input-ref]}]
+  (let [cursor            @(re-frame/subscribe [:chat/cursor])
+        mentionable-users @(re-frame/subscribe [:chats/mentionable-users])]
+    [rn/view {:style (styles/text-input-wrapper)}
+     [rn/text-input
+      {:style                  (styles/text-input)
+       :ref                    text-input-ref
+       :maxFontSizeMultiplier  1
+       :accessibility-label    :chat-message-input
+       :text-align-vertical    :center
+       :multiline              true
+       :default-value          text-value
+       :editable               (not cooldown-enabled?)
+       :blur-on-submit         false
+       :auto-focus             false
+       :on-focus               #(set-active-panel nil)
+       :max-length             chat.constants/max-text-size
+       :placeholder-text-color (:text-02 @colors/theme)
+       :placeholder            (if cooldown-enabled?
+                                 (i18n/label :cooldown/text-input-disabled)
+                                 (i18n/label :t/type-a-message))
+       :underlineColorAndroid  :transparent
+       :auto-capitalize        :sentences
+       :selection
+       (when (and cursor platform/android?)
+         (clj->js
+          {:start cursor
+           :end   cursor}))
+
+       :on-selection-change
+       (fn [_]
+         (when cursor
+           (re-frame/dispatch [::mentions/clear-cursor])))
+
+       :on-change
+       (fn [args]
+         (let [text (.-text ^js (.-nativeEvent ^js args))]
+           (on-text-change text)
+           (when platform/ios?
+             (re-frame/dispatch [::mentions/calculate-suggestions mentionable-users]))))
```

Comment:
i would suggest to use `debounce/debounce-and-dispatch` here, because if we'll dispatch on every symbol, input will be slow , we should calculate suggestions only when the user has stopped typing

Reply:
How can I verify that input will be slow? I mean I tried on average Android and it looks just fine.

Reply:
Just to clarify: in case if there is no "@" in text this is about one `if` in handler.

Reply:
i mean why would you want to flood the queue with events if you know you need only the latest one , so when you typing fast and when press send, your send event will be in the end of this queue of events, 400ms delay will do the job

Reply:
> i mean why would you want to flood the queue with events

because I don't want to have 400ms delay unless it is really necessary? :)

Reply:
from my experience its better to use it , better for performance, but its up to you

Reply:
I will test it more carefully in the next PR which will be about highlighting mention in the text input and then we will decide if this is necessary.